### PR TITLE
fix(behavior_velocity_planner_common): sync activation when safety status is set

### PR DESCRIPTION
**Pull request by rej55**
## Description

Before this change, activation status is not updated in the cycle that safety status changed.
In this PR, it enables to sync activation status with current safety status.

## Tests performed

[TIER IV INTERNAL TEST](https://evaluation.tier4.jp/evaluation/reports/e0fd3032-83de-57f5-bcbe-98acdfb88349?project_id=prd_jt)

Planning simulator.

Before:

image_720

After:

Screenshot from 2023-11-28 15-30-22

## Interface changes

Not applicable.

## Effects on system behavior

Activation status will be updated in the cycle that safety status updated.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_velocity_planner_common/include/behavior_velocity_planner_common/scene_module_interface.hpp [hunk 1]
```diff
@@ -106,8 +108,18 @@ class SceneModuleInterface
...
+    safe_ = safe;
+    syncActivation();
+  }
   void setDistance(const double distance) { distance_ = distance; }
+  void syncActivation()
+  {
+    if (!rtc_enabled_) {
+      setActivation(isSafe());
+    }
+  }
```
Comment (kyoichi-sugahara) [inline planning/behavior_velocity_planner_common/include/behavior_velocity_planner_common/scene_module_interface.hpp:122]: [NITS] super minor suggestion you can ignore
```suggestion
    if (!rtc_enabled_) {
      syncActivation()
    }
  }
  void setDistance(const double distance) { distance_ = distance; }
  void syncActivation() { setActivation(isSafe()); }
```

**Review by kyoichi-sugahara — APPROVED** [review]
LGTM

**Review by taikitanaka3 — APPROVED** [review]
LGTM

**Review by tkimura4 — APPROVED** [review]

**Review by shmpwk — APPROVED** [review]

File: planning/behavior_velocity_planner_common/include/behavior_velocity_planner_common/scene_module_interface.hpp [hunk 2]
```diff
@@ -106,8 +108,15 @@ class SceneModuleInterface
   boost::optional<int> first_stop_path_point_index_;
   VelocityFactorInterface velocity_factor_;
 
-  void setSafe(const bool safe) { safe_ = safe; }
+  void setSafe(const bool safe)
+  {
+    safe_ = safe;
+    if (!rtc_enabled_) {
+      syncActivation()
```
[the remark this answers is not in the document]
Comment (tkimura4) [reply planning/behavior_velocity_planner_common/include/behavior_velocity_planner_common/scene_module_interface.hpp:115]: @rej55

**Pull request merged**